### DISCUSSION TITLE
For Angular 16 and PRIMENG 16.9.0 and Prime NG California theme, the Menu does not working when using as Popup menu in p-table.

### DISCUSSION THREAD

**Discussion by nazrulcsebd** [primefaces/community#536] [PrimeNG Templates] [answered]
The angular version is 15 and the PRIME NG version is 15.2.0. We are using California Theme. We are using menu as popup in the p-table, it's working fine. But In Angular 16 or 17 the menu is not working,  its loaded and loaded. Please give us a solution or fix it. We need to version change very urgently. We are also implementing the popup menu to the pure California theme 16 version, the same problem occurred.

The code is given here. If i provide static menu items data then it working fine.

`<p-table id="tblAppPackage" [value]="packageGridObj" styleClass="p-datatable-gridlines">
    <ng-template pTemplate="header">
        <tr>
            <th style="width: 5%">Sr. No.</th>
            <th style="width: 25%">TNo., <br>Package Description</th>
            <th style="width: 10%">Type</th>
            <th style="width: 12%">Cost <br>(In BDT)</th>
            <th style="width: 12%">Estimated Cost</th>
            <th style="width: 12%">Proposal ID</th>
            <th style="width: 10%">Proposal Status</th>
            <th style="width: 8%">Status</th>
            <th style="width: 6%">Action</th>
        </tr>
    </ng-template>
    <ng-template pTemplate="body" let-package let-ri="rowIndex">
        <tr>
            <td class="text-center" style="width: 5%;">{{ startRowIndex + ri + 1 }}</td>
            <td style="width: 25%">{{package?.packageNo}} <p-tag styleClass="me-2"
                    [rounded]="true">{{package?.procurementNature}}</p-tag> <p-tag [rounded]="true"
                    styleClass="butter-bg">{{package.procurementType}}</p-tag> <br>
                {{ package?.packageDesc }}
            </td>
            <td class="text-center" style="width: 10%">{{package?.appType}}</td>
            <td class="text-center" style="width: 12%">{{package?.pkgEstCost}}</td>
            <td class="text-center" style="width: 12%;">
                <span id="btnUploadEstimatedCost" *ngIf="isEstimatedCostDocumentUploadable(package)"
                    (click)="goToUploadEstimatedCost(package)" class="border-ash">Click here, to upload Estimated
                    Cost<img src="assets/layout/images/svg/upload-icon.svg" /></span>
                <span id="btnViewEstimatedCost" *ngIf="isEstimatedCostDocumentViewable(package)"
                    (click)="goToViewUploadEstimatedCost(package)">View</span>
                <span *ngIf="!isEstimatedCostDocumentUploadable(package) && !isEstimatedCostDocumentViewable(package)">
                    - </span>
            </td>
            <td class="text-center" style="width: 12%">
                <span *ngIf="!package?.tenderId"> - </span>
                <span *ngIf="package?.tenderId">{{package.tenderId}}</span>
            </td>
            <td style="width: 10%">
                {{ package.tenderStatus }}
            </td>
            <td style="width: 8%" class="text-center">
                <span *ngIf="checkPackageStatus(package?.status)" style="color: red; font-weight: bold;">{{
                    checkPackageStatusValue(package) }}</span>
                <span *ngIf="!checkPackageStatus(package?.status)">{{ package?.status }}</span>
            </td>
            <td class="text-center" style="width: 6%">
                <p-menu #menu [popup]="true" [model]="itemsOptions" appendTo="body"></p-menu>
                <button type="button" pButton icon="pi pi-chevron-down" label="Options"
                    (click)="menu.toggle($event)" style="width:auto"></button>            </td>
        </tr>
    </ng-template>
</p-table>`

Ts file code

` itemsOptions: MenuItem[] = [];
ngOnInit(): void {

    this.itemsOptions = [
      {
          label: 'Save', icon: 'pi pi-fw pi-check'
      },
      {
          label: 'Update', icon: 'pi pi-fw pi-refresh'
      },
      {
          label: 'Delete', icon: 'pi pi-fw pi-trash'
      },
      {
          separator: true
      },
      {
          label: 'Home', icon: 'pi pi-fw pi-home'
      },
  ];
}
`

But if I generate menu items dynamically then the problem is occured. For same example the the code is given below

`<p-table id="tblAppPackage" [value]="packageGridObj" styleClass="p-datatable-gridlines">
    <ng-template pTemplate="header">
        <tr>
            <th style="width: 5%">Sr. No.</th>
            <th style="width: 25%">TNo., <br>Package Description</th>
            <th style="width: 10%">Type</th>
            <th style="width: 12%">Cost <br>(In BDT)</th>
            <th style="width: 12%">Estimated Cost</th>
            <th style="width: 12%">Proposal ID</th>
            <th style="width: 10%">Proposal Status</th>
            <th style="width: 8%">Status</th>
            <th style="width: 6%">Action</th>
        </tr>
    </ng-template>
    <ng-template pTemplate="body" let-package let-ri="rowIndex">
        <tr>
            <td class="text-center" style="width: 5%;">{{ startRowIndex + ri + 1 }}</td>
            <td style="width: 25%">{{package?.packageNo}} <p-tag styleClass="me-2"
                    [rounded]="true">{{package?.procurementNature}}</p-tag> <p-tag [rounded]="true"
                    styleClass="butter-bg">{{package.procurementType}}</p-tag> <br>
                {{ package?.packageDesc }}
            </td>
            <td class="text-center" style="width: 10%">{{package?.appType}}</td>
            <td class="text-center" style="width: 12%">{{package?.pkgEstCost}}</td>
            <td class="text-center" style="width: 12%;">
                <span id="btnUploadEstimatedCost" *ngIf="isEstimatedCostDocumentUploadable(package)"
                    (click)="goToUploadEstimatedCost(package)" class="border-ash">Click here, to upload Estimated
                    Cost<img src="assets/layout/images/svg/upload-icon.svg" /></span>
                <span id="btnViewEstimatedCost" *ngIf="isEstimatedCostDocumentViewable(package)"
                    (click)="goToViewUploadEstimatedCost(package)">View</span>
                <span *ngIf="!isEstimatedCostDocumentUploadable(package) && !isEstimatedCostDocumentViewable(package)">
                    - </span>
            </td>
            <td class="text-center" style="width: 12%">
                <span *ngIf="!package?.tenderId"> - </span>
                <span *ngIf="package?.tenderId">{{package.tenderId}}</span>
            </td>
            <td style="width: 10%">
                {{ package.tenderStatus }}
            </td>
            <td style="width: 8%" class="text-center">
                <span *ngIf="checkPackageStatus(package?.status)" style="color: red; font-weight: bold;">{{
                    checkPackageStatusValue(package) }}</span>
                <span *ngIf="!checkPackageStatus(package?.status)">{{ package?.status }}</span>
            </td>
            <td class="text-center" style="width: 6%">
                <p-menu #menu [popup]="true" [model]="getPackageOptionMenuItem(package)" appendTo="body"></p-menu>
                <button type="button" pButton icon="pi pi-chevron-down" label="Options"
                    (click)="menu.toggle($event)" style="width:auto"></button>            </td>
        </tr>
    </ng-template>
</p-table>`

The ts file code is given below

`
  getPackageOptionMenuItem(packageItem: any) {
    let items: MenuItem[] = [];

    let currPackageSideLink = this.packageSideLinkObj?.filter(pksl => pksl.packageId == packageItem.packageId)[0];

    if (currPackageSideLink) {

      if (currPackageSideLink.processFileInWkf) {
        let processFileInWorkflow: MenuItem = {
          label: 'Process file in Workflow', id: 'linkProcessFileWorkflow', icon: 'pi pi-folder-open', command: (event) => {
            this.fileProcessingForAppWorkflow();
          }
        };
        items.push(processFileInWorkflow);
      }

      if (currPackageSideLink.viewPkg) {
        let viewPackage: MenuItem = {
          label: 'View', id: 'linkViewPackageDetails', icon: 'pi pi-eye', command: (event) => {
            this.goToPackageDetails(packageItem);
          }
        };
        items.push(viewPackage);
      }

      if (currPackageSideLink.publish) {
        let appPackagePublish: MenuItem = {
          label: 'Publish', id: 'linkAppPackagePublish', icon: 'pi pi-check-circle', command: (event) => {
            this.goToAppPackagePublish(packageItem);
          }
        };
        items.push(appPackagePublish);
      }

      
    }

    return items;

  }
`

![angular 15 primeng 15 2](https://github.com/primefaces/community/assets/3226667/6e9f6388-2d9d-49fa-b38d-651994758872)

When the angular version is 16 and PRIME NG version is 16.9.0. We are using California Theme. The menu is not 
working.  Its load and load frequently, see in left pane (attached image) detected in the browser. For all browser it show same problem.

![menu load frequently on p table angular 17](https://github.com/primefaces/community/assets/3226667/e61bb1b1-e114-4cc2-a842-95abcf53ebe6)

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

PrimeNG versions are sync with Angular versions, for example 

Angular 15 -> PrimeNG 15

Angular 16 -> PrimeNG 16

And Angular does not fully support all the previous versions. There may be some deprecations and breaking changes between versions.  Please use the synced versions.

**Comment by nazrulcsebd** [comment]
Hi,
Prime NG version is 16.0.9 and angular version is 16.0.2. Version is same,
here may be other problem
